app/assets/javascripts/oxalis/model/sagas/save_saga.js

```diff
@@ -175,6 +175,20 @@ export function compactUpdateActions(updateActionsBatches: Array<Array<UpdateAct
   });
 
 
+  // This part of the code detects deleted trees.
+  // Instead of sending deleteNode/deleteEdge update actions for all nodes of a deleted tree,
+  // just one deleteTree update action is sufficient for the server to delete the tree.
+  // As the deleteTree update action is already part of the update actions if a tree is deleted,
+  // all corresponding deleteNode/deleteEdge update actions can simply be removed.
+  result = result.map((batch) => {
+    // TODO: Remove the check in map once Flow recognizes that the result of the filter contains only deleteTree update actions
+    const deletedTreeIds = batch.filter(ua => ua.action === "deleteTree").map(ua => ua.action === "deleteTree" && ua.value.id);
```

Comment:
Perhaps I am missing something here but the `&&` looks weird. Shouldn't the result of a boolean and be a boolean itself? Hence, the `map` returns booleans? 😕